File: SwiatZwierzat_Java/Lion.java
public class Lion extends Mammal implements Carnivore {  //*dziedziczenie* extends uzywamy podczas dziedziczenia klas. Zapobiega to powielaniu tych samych cech obiektow. Dzieki temu mozemy
    //możemy stworzyć oddzielną klasę, czyli Animal i odziedziczyć po niej zestaw cech które ma każde zwierzę
    //z kolei implements ma podobne znaczenie do extends jednak w tym przypadku oznajmiamy ze de facto bedziemy wykorzystywac dany interfejs,
    //korzystac i zapisywac cialo metod z interfejsu
    Lion(){
        super();  //za pomoca slowa super w tym przypadku wywolujemy konstruktor klasy, po ktorej dziedziczymy
        System.out.println("Jestem Lwem");
    }

    @Override
    public void eatMeat() {
        System.out.println("Jestem miesozerca. Poluje glownie na antylopy i gazele.");
    }

    @Override
    public void presentYourself() {
        System.out.println("Mam na imie " + getName() + " oraz zyje tu na sawannie juz od " + getAge() + " lat");
        eatMeat();
    }
}

File: SwiatZwierzat_C++/Lion.cpp
#pragma once

#include "Mammal.cpp"
#include "Carnivore.cpp"
class Lion: public Mammal, public Carnivore  {  //*dziedziczenie* slowo public oznacza, że dziedziczone elementy (np. zmienne lub funkcje) mają taką widoczność jak w klasie bazowej, oprocz private w bazowej. Zapobiega to powielaniu tych samych cech obiektow. Dzieki temu mozemy
				//mozemy stworzyc oddzielna klase, czyli np Mammal i odziedziczyc po niej zestaw cech które ma kazdy ssak
public:

    Lion() : Mammal(){ //wywolujemy konstruktor klasy, po ktorej dziedziczymy, bo lew tez jest ssakiem
        cout << "Jestem Lwem" << endl;
    }

    void eatMeat() {
        cout << "Jestem miesozerca. Poluje glownie na antylopy i gazele." << endl;
    }

	void presentYourself() {
        cout << "Mam na imie " << getName() << " oraz zyje tu na sawannie juz od " << getAge() << " lat" << endl;
        eatMeat();
    }
};
